### PR TITLE
Warning fixes

### PR DESCRIPTION
**Pull request by kb1vc**
Fixes for compiler warnings from GCC 6.3.1  

Also added Doxygen "doc" target to produce limited Doxygen html pages.

### CODE REVIEW

File: src/lms7002m/mcu_dc_iq_calibration.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "mcu_programs.h"
 
-LIME_API const uint8_t mcu_program_lms7_dc_iq_calibration_bin[16384] = {
+// LIME_API // Is this really needed here?  It triggers some ugly warnings.
```
Comment (guruofquality) [inline src/lms7002m/mcu_dc_iq_calibration.cpp]: The macro should just be setting gcc visibility. In this case the pointer to the mcu program is used outside of the main library for the GUI. Whats the warning?

File: mcu_program/common_src/lms7002m_filters.c [hunk 2]
```diff
@@ -170,12 +170,14 @@ uint8_t TuneRxFilterSetup(const float_type rx_lpf_IF)
...
+    if(g_pga_rbb == 31) {
         Modify_SPI_Reg_bits(G_PGA_RBB, 22);
-    else
+    }
+    else {
         Modify_SPI_Reg_bits(G_PGA_RBB, g_pga_rbb);
+    }
 
-	{
+    {
```
Comment (guruofquality) [inline mcu_program/common_src/lms7002m_filters.c]: This indentation and brackets was definitely unclear. just to double check, @IgnasJarusevicius was this the original intention?

Comment (IgnasJarusevicius) [inline mcu_program/common_src/lms7002m_filters.c]: I am certain that it was. It was just bad indentation. The brackets are there to limit scope of variable to help compiler optimize memory usage. You can see this a lot in the code used by MCU as it has very limited resources.

File: mcu_program/common_src/lms7002m_calibrations.c [hunk 3]
```diff
@@ -84,13 +84,15 @@ void LoadDC_REG_TX_IQ()
     FlipRisingEdge(TSGDCLDQ_TXTSP);
 }
 
-static void Dummy()
+#if 0 // nobody needs this right now.
```
Comment (IgnasJarusevicius) [inline mcu_program/common_src/lms7002m_calibrations.c]: I think it is used when compiling calibrations image for 8051 MCU. Also, I think at least some of the warnings in C files located in 'mcu_program' folder might be intentional and meant to deal with memory/performance/timing constraints in MCU.

**Comment by kb1vc** [comment]
Ignas,



Having written a few hundred thousand lines of code over the past 40 years, and managed some

very very large software and hardware projects, I can say categorically that warnings in compiler

output are a danger sign that should never be ignored.  I have never worked in a place that did

production code that would tolerate warnings in the nightly build, never mind in released code.



So, to your problematic mcu_dc_iq_calibration.cpp code.  How do you know that it is not a problem?

You know because you have looked at it carefully, consulted with at least one responsible and diligent

colleague, and thoroughly understand the meaning of the warning.  I'm sure you've done that already,

as otherwise you'd be shipping code that might be broken, and we all want to do our best.



Now that you've reviewed the warning and deemed that particular instance of the warning as

benign, you don't want to have to do this again, each time the build complains.  You either fix the

expression in the source, if that is possible and prudent, or you turn the warning off FOR JUST THAT

INSTANCE OF THE PROBLEM.  Turning off a warning for the entire code base is like putting black tape

over warning lights or not going to the doctor because you're afraid of what you'll find.  It is irresponsible,

and unprofessional.



Here's a diff showing how to turn off the warning for just that one instance in mcu_dc_iq_calibration.cpp.

------------------------

$ git diff master src/lms7002m/mcu_dc_iq_calibration.cpp
diff --git a/src/lms7002m/mcu_dc_iq_calibration.cpp b/src/lms7002m/mcu_dc_iq_calibration.cpp
index 73366b5..ba58025 100644
--- a/src/lms7002m/mcu_dc_iq_calibration.cpp
+++ b/src/lms7002m/mcu_dc_iq_calibration.cpp
@@ -1,5 +1,6 @@
#include "mcu_programs.h"
-
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wattributes"
LIME_API const uint8_t mcu_program_lms7_dc_iq_calibration_bin[16384] = {
0x02,0x35,0xB4,0x12,0x3B,0x6F,0x7D,0xFF,0x7F,0x00,0x7E,0x04,0x12,0x3B,0x03,0x7F,
0x0F,0x7E,0x04,0x12,0x39,0x20,0xEF,0x78,0x02,0xC3,0x33,0xCE,0x33,0xCE,0xD8,0xF9,
@@ -1026,3 +1027,4 @@ LIME_API const uint8_t mcu_program_lms7_dc_iq_calibration_bin[16384] = {
0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
};
+#pragma GCC diagnostic pop

----------------------------



There are similar pragmas for the various microsoft compilers, and one might even consider a

language determined macro that could be built by CMake at configuration time. 



But if you really want to write code like a pro, avoid global warning switches, and fix all compiler

warnings and errors in the build.  The current version of LimeSuite is improving, but when I first

built it I saw more than 50 warnings, and at least two of them pointed to real bugs.   One of them was

in code you wrote, and it was a version of the "goto fail" bug.



matt





matt






Sent from iCloud

On Jun 03, 2017, at 12:58 PM, IgnasJarusevicius <notifications@github.com> wrote:



@IgnasJarusevicius commented on this pull request.

In mcu_program/common_src/lms7002m_calibrations.c:

> @@ -84,13 +84,15 @@ void LoadDC_REG_TX_IQ()
     FlipRisingEdge(TSGDCLDQ_TXTSP);
 }
 
-static void Dummy()
+#if 0 // nobody needs this right now.


I think it is used when compiling calibrations image for 8051 MCU. Also, I think at least some of the warnings in C files located in 'mcu_program' folder might be intentional and meant to deal with memory/performance/timing constraints in MCU.

**Pull request merged**